.gitattributes

```diff
@@ -0,0 +1 @@
+.dev/format.sh text eol=lf
```

Comment:
Do we need this file in the master?

Reply:
See last point of the list. 

As `sh`  has not a very helpful output, you are only getting this
```
/home/sunblack/dev/pcl/
: not found2: 
: not found8: 
format.sh: 11: local: not in a function
```

And as I needed a few hours to check that only the line endings are wrong I think it is better to force git to set LF line breaks there, before the next one will spend wasted time with it (I can remember I had this issues some years ago already with another script - I ask me always, why still common tools exists which requires a specific line break).

Reply:
Might I suggest some modification of your Windows git config
```
git config --global core.autocrlf false
git config --global core.eol lf
```
As for .gitattributes, the following is more general
```
* text=auto
```

Reply:
`autocrlf` can't be input or true because you're using it concurrently on 2 different OS

Reply:
This isn't really a good solution, as it effects all projects (and MSVC asking by default to convert LF to CRLF). So only alternative would be to add `*.sh text eol=lf` to a global `.gitattributes` (not sure if git supports it aside if global `.gitconfig`).

Nevertheless I'm prefer to have this in this repo, as my workflow may a bit unusual (Shared-Directory from Win to Linux), but you have the same issue with WSL. So I'm for having this into the repo or to exchange the `format.sh` script by a python or cmake script (I prefer last one), as `.sh` is really Linux specific, but we have also some Windows developer, which maybe want to run clang-format.

Reply:
Played a bit further with it: MSVC only ask in case there are misced line ending.